### ISSUE TITLE
Add support for Black Magick Output Device

### ISSUE THREAD

**Issue by ggarra13** [ggarra13/mrv2#164]
Currently, mrv2 has the source code for the UI for Blackmagic (and potentially others) output devices.  However the actual code has not been written as I don't have an actual device to test with (it currently costs u$1500).
Darby has written code to support them in tlRender, so porting it should be trivial.

**Comment by SheepDomination** [comment]
What do you mean, I own a BMPCC4K that I have yet to learn; what is it that you are after ?

**Comment by ggarra13** [comment]
BlackMagic Decklinks Cinema Capture supports cards that can play at 4K and 8K more smoothly than your computer:

https://www.amazon.com/Blackmagic-Design-Decklink-Cinema-Capture/dp/B07BBQ7MPC/ref=sr_1_9?crid=3B7RFUA5MZDLM&keywords=black+magic+design+card&qid=1702490804&sprefix=black+magic+design+card%2Caps%2C267&sr=8-9

I do not own one, and until yesterday imports in my country were frozen and expensive, even though for US they are now relatively cheap.

**Comment by ggarra13** [comment]
If you want to take advantage of a Decklink card *now*, I recommend you look into (for now, compiling from source) Darby's new viewer called tlplay-gl which ships with tlRender's source code.

**Comment by SheepDomination** [comment]
I have to look into these cards, I don't personally own one as to what their advantages are over feeding the video feed into the computer directly.

**Comment by ggarra13** [comment]
No.  I chose to use NDI instead of Blackmagic as they have a more solid API that does not change on each hardware release.  It also does not necessary require buying new hardware for 4K or HDR support if you already have those.  The use for Blackmagic now is mainly for stereo and 8K support.

If you do want it supported, you can become a sponsor and hire me to add it for you/your company personally.
